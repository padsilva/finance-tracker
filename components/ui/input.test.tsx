import { render } from "@testing-library/react";

import { Input } from "./input";

describe("Input Component", () => {
  it("should render default input correctly", () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });

  it("should render different input types correctly", () => {
    const { container } = render(
      <>
        <Input type="text" placeholder="Text input" />
        <Input type="password" placeholder="Password input" />
        <Input type="email" placeholder="Email input" />
        <Input type="number" placeholder="Number input" />
        <Input type="tel" placeholder="Telephone input" />
        <Input type="search" placeholder="Search input" />
        <Input type="url" placeholder="URL input" />
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with custom className", () => {
    const { container } = render(
      <Input className="custom-class" placeholder="Custom styled input" />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with different states", () => {
    const { container } = render(
      <>
        <Input disabled placeholder="Disabled input" />
        <Input readOnly placeholder="Readonly input" />
        <Input required placeholder="Required input" />
        <Input aria-invalid placeholder="Invalid input" />
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with different sizes via className", () => {
    const { container } = render(
      <>
        <Input className="h-8 text-sm" placeholder="Small input" />
        <Input className="h-10" placeholder="Default input" />
        <Input className="h-12 text-lg" placeholder="Large input" />
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with prefix and suffix", () => {
    const { container } = render(
      <>
        <div className="relative">
          <span className="absolute left-3 top-1/2 -translate-y-1/2">$</span>
          <Input className="pl-7" placeholder="Amount" type="number" />
        </div>
        <div className="relative">
          <Input className="pr-12" placeholder="Weight" type="number" />
          <span className="absolute right-3 top-1/2 -translate-y-1/2">kg</span>
        </div>
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with form attributes", () => {
    const { container } = render(
      <>
        <Input
          name="username"
          form="test-form"
          autoComplete="username"
          placeholder="Username"
        />
        <Input
          name="password"
          form="test-form"
          autoComplete="current-password"
          type="password"
          placeholder="Password"
        />
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with ARIA attributes", () => {
    const { container } = render(
      <>
        <Input
          aria-label="Search"
          aria-describedby="search-desc"
          placeholder="Search..."
        />
        <Input
          aria-required="true"
          aria-invalid="true"
          aria-errormessage="error-msg"
          placeholder="Required field"
        />
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with data attributes", () => {
    const { container } = render(
      <Input
        data-testid="test-input"
        data-cy="cypress-input"
        placeholder="Test input"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
