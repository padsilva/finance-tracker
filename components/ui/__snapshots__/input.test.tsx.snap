// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render default input correctly 1`] = `
<div>
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
  />
</div>
`;

exports[`Input Component should render different input types correctly 1`] = `
<div>
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Text input"
    type="text"
  />
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Password input"
    type="password"
  />
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Email input"
    type="email"
  />
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Number input"
    type="number"
  />
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Telephone input"
    type="tel"
  />
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Search input"
    type="search"
  />
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="URL input"
    type="url"
  />
</div>
`;

exports[`Input Component should render with ARIA attributes 1`] = `
<div>
  <input
    aria-describedby="search-desc"
    aria-label="Search"
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Search..."
  />
  <input
    aria-errormessage="error-msg"
    aria-invalid="true"
    aria-required="true"
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Required field"
  />
</div>
`;

exports[`Input Component should render with custom className 1`] = `
<div>
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm custom-class"
    placeholder="Custom styled input"
  />
</div>
`;

exports[`Input Component should render with data attributes 1`] = `
<div>
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    data-cy="cypress-input"
    data-testid="test-input"
    placeholder="Test input"
  />
</div>
`;

exports[`Input Component should render with different sizes via className 1`] = `
<div>
  <input
    class="flex w-full rounded-md border border-input bg-transparent px-3 py-1 shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm h-8 text-sm"
    placeholder="Small input"
  />
  <input
    class="flex w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm h-10"
    placeholder="Default input"
  />
  <input
    class="flex w-full rounded-md border border-input bg-transparent px-3 py-1 shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm h-12 text-lg"
    placeholder="Large input"
  />
</div>
`;

exports[`Input Component should render with different states 1`] = `
<div>
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    disabled=""
    placeholder="Disabled input"
  />
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Readonly input"
    readonly=""
  />
  <input
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Required input"
    required=""
  />
  <input
    aria-invalid="true"
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    placeholder="Invalid input"
  />
</div>
`;

exports[`Input Component should render with form attributes 1`] = `
<div>
  <input
    autocomplete="username"
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    form="test-form"
    name="username"
    placeholder="Username"
  />
  <input
    autocomplete="current-password"
    class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    form="test-form"
    name="password"
    placeholder="Password"
    type="password"
  />
</div>
`;

exports[`Input Component should render with prefix and suffix 1`] = `
<div>
  <div
    class="relative"
  >
    <span
      class="absolute left-3 top-1/2 -translate-y-1/2"
    >
      $
    </span>
    <input
      class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm pl-7"
      placeholder="Amount"
      type="number"
    />
  </div>
  <div
    class="relative"
  >
    <input
      class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm pr-12"
      placeholder="Weight"
      type="number"
    />
    <span
      class="absolute right-3 top-1/2 -translate-y-1/2"
    >
      kg
    </span>
  </div>
</div>
`;
